core/src/main/java/ai/timefold/solver/core/impl/constructionheuristic/decider/ConstructionHeuristicDecider.java

```diff
@@ -155,12 +157,12 @@ protected void pickMove(ConstructionHeuristicStepScope<Solution_> stepScope) {
 
     protected <Score_ extends Score<Score_>> void doMove(ConstructionHeuristicMoveScope<Solution_> moveScope) {
         InnerScoreDirector<Solution_, Score_> scoreDirector = moveScope.getScoreDirector();
-        scoreDirector.doAndProcessMove(moveScope.getMove(), assertMoveScoreFromScratch, score -> {
-            moveScope.setScore(score);
-            forager.addMove(moveScope);
-        });
+        var score = scoreDirector.executeTemporaryMove(moveScope.getMove(), assertMoveScoreFromScratch);
+        moveScope.setScore(score);
+        forager.addMove(moveScope);
         if (assertExpectedUndoMoveScore) {
-            scoreDirector.assertExpectedUndoMoveScore(moveScope.getMove(),
+            var undoMoveToString = "Undo(%s)".formatted(moveScope.getMove());
```

Comment:
This variable seems off to me. I think it should be within the scope of `scoreDirector.assertExpectedUndoMoveScore`.